secure/key-manager/src/counters.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_secure_push_metrics::{define_counters, Counter};
+use once_cell::sync::Lazy;
+use std::sync::Arc;
+
+// Metrics for the key manager. Uses the "libra_key_manager" prefix for all counters.
+define_counters![
+    "libra_key_manager",
+    (
+        full_consensus_rotations: Counter,
+        "counts the number of full consensus key rotations performed by the key manager"
+    ),
+    (
+        rotation_transaction_resubmissions: Counter,
+        "counts the number of times the key manager had to resubmit a rotation transaction to the blockchain"
+    ),
+    (
+        no_actions_required: Counter,
+        "counts the number of times the key manager determined that no actions were required"
+    ),
+    (
+        reconfiguration_waits: Counter,
+        "counts the number of times the key manager had to wait for a reconfiguration event"
+    ),
+    (
+        sleeps: Counter,
+        "counts the number of times the key manager went to sleep"
+    ),
+];
+
+pub static COUNTERS: Lazy<Arc<Counters>> = Lazy::new(|| Arc::new(Counters::new()));
```

Comment:
in general it isn't clear if we are referring to the time between rotations or the time since the process began.... also counters start at zero if restarted... some food for thought

cc: @mgorven and @sherry-x for any feedback on useful counters

Reply:
All metrics are since the process began (no counters are reset during execution). In my mind, the execution model is that if the key manager stops executing, something must have gone wrong, so it makes sense that all counters should be restarted. One could infer the time between reconfigurations by watching the metrics update.